controllers/mysqlcluster_controller.go

```diff
@@ -260,6 +296,101 @@ func generateRandomBytes(n int) ([]byte, error) {
 	return ret, nil
 }
 
+func mergeMycnf(cnf1, cnf2 map[string]string) {
+	for k, v := range cnf2 {
+		cnf1[normalizeConfKey(k)] = v
+	}
+}
+
+func normalizeConfKey(k string) string {
+	return strings.ReplaceAll(k, "-", "_")
+}
+
+func (r *MySQLClusterReconciler) createOrUpdateConfigMap(ctx context.Context, log logr.Logger, cluster *mocov1alpha1.MySQLCluster) error {
+	cm := &corev1.ConfigMap{}
+	cm.SetNamespace(cluster.Namespace)
+	cm.SetName(cluster.Name)
+
+	op, err := ctrl.CreateOrUpdate(ctx, r.Client, cm, func() error {
+		cm.Labels = map[string]string{
+			appNameKey: fmt.Sprintf("%s-%s", appName, cluster.Name),
+		}
+
+		conf := make(map[string]mysqlConf)
+		conf["mysqld"] = make(mysqlConf)
+		mergeMycnf(conf["mysqld"], defaultMycnf)
```

Comment:
This looks dangerous because of the necessities of `make` maps.
Why don't you abstract it further?